### DISCUSSION TITLE
Creating custom column that updates `record`

### DISCUSSION THREAD

**Discussion by eithed** [filamentphp/filament#10267] [Help]
I'm trying to create a custom column; let's, for the sake of simplicity, assume that I want to replicate functionality of TextInput column. As such I define my column within my Table schema declaration:
```
ViewColumn::make('status')
   ->view('template')
```
and additionally create this method on my Table class:
```
public function updateState(string $className, string $id, array $data)
{
    $model = $className::find($id);

    dd($model, $data)
}
```

Now within my `template` I do following:
```
<input wire:change="updateState('{{addslashes(get_class($getRecord()))}}', '{{$getRecord()->id}}', {status: event.target.value})"/>
```

As such I've defined a text input that sends class name of record, its id and value of the field to backend; this then gets reconstituted using `updateState` method to an instance of a model and then finally I can use a Model + data that was sent from frontend.

Is there a way to tidy up this code - what I'm finding ugly is:
1. my ViewColumn + updateState exists within scope of my Table component; I want for this functionality to exist within scope of Column component
2. I'm having to transform my record to classname and id and then back to a model instance

To resolve 1 I've tried creating an instance of `StatusColumn` class that extends `Column` and move `updateState` there; this doesn't work and I'm getting error that `updateState` is not defined on my Table.

I've no clue on how to resolve 2

Thanks!

**Comment by ArtMin96** [comment]
If I understood correctly

[Column actions](https://filamentphp.com/docs/3.x/tables/columns/getting-started#running-actions)

```php
ViewColumn::make('status')
    ->action(function (YourRecord $record) {
        //...
    })
```
?

**Reply by eithed** [reply]
Thanks for the suggestion! Unfortunately this will cause an action to trigger as soon as column is clicked.

I'd like to perform an action on custom trigger - in this instance, when an `<input>` value has changed

**Comment by Joemires** [comment]
Hey @eithed 
Where you able to find a solution around this?
Tho I've been able to create a little something to solve this issue challenge
